Mathlib/Init/Data/Nat/Bitwise.lean

```diff
@@ -186,41 +186,51 @@ def shiftl' (b : Bool) (m : ℕ) : ℕ → ℕ
   | n + 1 => bit b (shiftl' b m n)
 #align nat.shiftl' Nat.shiftl'
 
-/-- `shiftl m n` produces a natural number whose binary representation
-  is obtained by left-shifting the binary representation of `m` by `n` places -/
-def shiftl : ℕ → ℕ → ℕ :=
-  shiftl' false
-#align nat.shiftl Nat.shiftl
+@[simp]
+theorem shiftLeft_eq_shiftl'_false : ∀ n, shiftl' false m n = m <<< n
+  | 0 => rfl
+  | n + 1 => by
+    have : 2 * (m * 2^n) = 2^(n+1)*m := by
+      rw [Nat.mul_comm, Nat.mul_assoc, ← pow_succ]; simp
+    simp [shiftl', bit_val, shiftLeft_eq_shiftl'_false, this]
 
 @[simp]
-theorem shiftl_zero (m) : shiftl m 0 = m :=
-  rfl
-#align nat.shiftl_zero Nat.shiftl_zero
+lemma bit_0 (b : Bool) : Nat.bit b 0 = b.toNat := by
+  cases' b <;> simp
 
 @[simp]
-theorem shiftl_succ (m n) : shiftl m (n + 1) = bit0 (shiftl m n) :=
-  rfl
-#align nat.shiftl_succ Nat.shiftl_succ
+lemma bit_1 (b : Bool) : Nat.bit b 1 = 2 + b.toNat:= by
+  cases' b <;> simp
+
+@[simp]
+lemma shiftLeft_eq' {m n : Nat} : shiftLeft m n = m <<< n := rfl
```

Comment:
Can you add a comment explaining why this name is primed?

Reply:
Sure, it's because ```shiftLeft_eq``` is already a lemma in ```Std.Data.Nat.Lemmas``` and it's also ```simp```ed (I don't think it should be simped either).